File: be/src/main/java/com/example/petclinic/dto/request/appointment/AppointmentCreation.java
package com.example.petclinic.dto.request.appointment;

import com.example.petclinic.entity.ServiceClinic;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import java.util.List;
import java.sql.Timestamp;

@Data
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class AppointmentCreation {
    String userId;
    String doctorId;
    List<ServiceClinic> services;
    Timestamp appointmentTime;
    int status;
}

File: ung_dung_thu_y/lib/dto/appointment/appointment_creation.dart
import 'package:ung_dung_thu_y/dto/service/services_get_dto.dart';

class AppointmentCreation {
  final String doctorId;
  final List<ServicesGetDto> services;
  final String appointmentTime;
  final int? status;
  AppointmentCreation({
    required this.doctorId,
    required this.services,
    required this.appointmentTime,
    this.status,
  });
  Map<String, dynamic> toJson() {
    return {
      'doctorId': doctorId,
      'services': services.map((s) => s.toJson()).toList(),
      'appointmentTime': appointmentTime,
      'status': status ?? 0,
    };
  }
}
